### DISCUSSION TITLE
Too many colors in the HTML output

### DISCUSSION THREAD

**Discussion by ahamlat** [async-profiler/async-profiler#728] [Q&A] [answered]
Hi there,
I was wondering if anyone else experienced this kind of output in the HTML file 

<img width="1715" alt="image" src="https://user-images.githubusercontent.com/5099602/226344010-b75dd856-74f6-4c27-b23c-794c67856b1b.png">

I had this output with a Java application running inside a Docker container. I just changed the docker image and I have a different output, the one I used to have, which I prefer by the way.
<img width="1715" alt="image" src="https://user-images.githubusercontent.com/5099602/226344371-25764574-6ddf-45d7-9e39-45c70fd15aed.png">

It is pretty hard for me to read the first one, is there any configuration or any explanation for this output ?

Thanks.

**Comment by apangin** [comment, marked as answer]
It is OpenJ9 VM on the first image. Unlike HotSpot JVM (on the second image), OpenJ9 has the API to obtain information about compilation status of each method in a stack trace. Async-profiler uses this API to provide detailed information on frame types:
 - Pale green color denotes methods running in the interpreter.
 - Bright green are JIT compiled methods.
 - Aqua color stands for inlined frames (method that do not have a separate frame on the stack, but belong to the frame of the caller method).

For HotSpot JVM, async-profiler can display frame types only for the topmost frames.

Distinguishing frame types with different colors is a demanded feature, I did not imagine this could be inconvient for users and thus did not provide an option to opt-out. However, it is possible to produce a simplifed Flame Graph by dumping results in an intermediate `collapsed` format and then converting it with a `FlameGraph` tool bundled with async-profiler:

```
./profile.sh ... -f out.collapsed <pid>
java -cp converter.jar FlameGraph out.collapsed out.html
```

**Reply by ahamlat** [reply]
Thanks @apangin, this is very helpful. I was actually comparing OpenJ9 performance against HotSpot. It was inconvenient for me because I wanted to compare the two outputs, but with your explanation, the colours may help also to find performance optimisations.
